public/app/plugins/datasource/loki/querybuilder/components/LokiQueryBuilderOptions.tsx

```diff
@@ -153,4 +141,29 @@ function getCollapsedInfo(
   return items;
 }
 
+export async function makeStatsRequest(
+  datasource: LokiDatasource,
+  timerange: TimeRange,
+  prevTimerange: TimeRange | undefined,
+  query: string,
+  prevQuery: string | undefined,
+  setQueryStats: React.Dispatch<React.SetStateAction<QueryStats | undefined>>
+) {
+  if (
+    query === prevQuery &&
+    timerange.raw.from === prevTimerange?.raw.from &&
+    timerange.raw.to === prevTimerange?.raw.to
+  ) {
+    return;
```

Comment:
This is nit/personal preference: I am wondering if we can just encapsulate this into `shouldRunStatsRequest` and check it outside of this function? And if the result is no, then we just want call this.